app/src/main/java/com/nutomic/syncthingandroid/util/DefaultHashMap.java

```diff
@@ -0,0 +1,13 @@
+package com.nutomic.syncthingandroid.util;
+
+import java.util.HashMap;
+
+/**
+ * Class implementing a HashMap with a getOrDefault interface.
+ * HashMap requires at least API level >= 24 to support getOrDefault natively.
+ */
+public class DefaultHashMap<K,V> extends HashMap<K,V> {
+    public V getOrDefault(Object k, V defaultValue) {
+        return containsKey(k) ? this.get(k) : defaultValue;
+    }
+}

```

Comment:
So I am pretty sure Java 8 has this out of the box. Where is the reference stating that this requires API level 24?

Reply:
Just remove DefaultFolderMap.java and try to build the project with native HashMap in place. Lint will tell you that you need to raise the API level. (I posted the message above in my comment)